source/braille.py

```diff
@@ -1441,9 +1441,123 @@ def _setCursor(self, info: textInfos.TextInfo):
 class ReviewTextInfoRegion(TextInfoRegion):
 
 	allowPageTurns=False
+	_currentSelection: textInfos.TextInfo | None = None
+	_currentScriptName: str | None = None
+	_withinSelection: bool = False
+
+	def _getSelection(self) -> textInfos.TextInfo:
+		"""Gets selection for use in update function.
+		:return: selection, reading unit from it, or review position when
+		there is no selection or showing selection is disabled
+		"""
+		if not config.conf["braille"]["showSelection"]:
+			self._currentSelection = None
+			return api.getReviewPosition().copy()
+		info: textInfos.TextInfo
+		try:
+			info = self.obj.makeTextInfo(textInfos.POSITION_SELECTION)
+		except (LookupError, RuntimeError):
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if info.isCollapsed:
+			# Cursor
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if (
+			self._currentSelection is None
+			or self._currentSelection.compareEndPoints(info, "startToStart") != 0
+			or self._currentSelection.compareEndPoints(info, "endToEnd") != 0
+		):
+			# Selection changed, update also review position
+			self._currentSelection = info.copy()
+			if self.obj.isTextSelectionAnchoredAtStart:
+				self._currentSelection.move(textInfos.UNIT_CHARACTER, -1, "end")
+			self._currentSelection.collapse(end=self.obj.isTextSelectionAnchoredAtStart)
+			api.setReviewPosition(self._currentSelection)
+			self._currentSelection = info.copy()
+			return info
+		# Selection unchanged
+		readingUnit: textInfos.TextInfo
+		readingUnit = api.getReviewPosition().copy()
+		readingUnit.expand(self._getReadingUnit())
+		# Selection may not contain whole reading unit.
+		if readingUnit.compareEndPoints(self._currentSelection, "startToStart") < 0:
+			readingUnit.setEndPoint(self._currentSelection, "startToStart")
+		if readingUnit.compareEndPoints(self._currentSelection, "endToEnd") > 0:
+			readingUnit.setEndPoint(self._currentSelection, "endToEnd")
+		if (
+			readingUnit.compareEndPoints(info, "startToEnd") >= 0
+			or readingUnit.compareEndPoints(info, "endToStart") <= 0

```

Comment:
New syntax on all these lines:
```suggestion
		if readingUnit.end > self._currentSelection.end:
			readingUnit.end = self._currentSelection.end
		if (
			readingUnit.start >= info.end
			or readingUnit.end <= info.start

```